### PR TITLE
Adding euphonic

### PR DESCRIPTION
**Pull request by ajjackson**
This is my first conda-forge package

- meta.yaml was generated using Grayskull and slightly tweaked to
  require numpy and clarify that this is _not_ a noarch

- [X] For some reason LICENSE is not currently included in the tarball,
  should try to get that fixed that in the main Euphonic repo.
- [X] I will check if other project members are willing to be listed as maintainer. It seems wise to have more than one.

I don't think there are issues around static library linking/licensing, but am not familiar with all the nuances around that. This is a Python package with C extensions that use Numpy and OpenMP.

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by rebeccafair** [comment]
Confirming I'm happy to be a maintainer :)

**Comment by ajjackson** [comment]
Linux is building! Windows is blowing up on a `git clone` while building the recipe. Maybe that will fix itself when switching back to the tarball source. (Currently using a Github branch so that upstream things can be fixed.)

**Comment by ajjackson** [comment]
@conda-forge-admin, please ping team

File: recipes/euphonic/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  commands:
+    - pip check
+    - euphonic-dispersion --help
+    - euphonic-dos --help
+    - euphonic-optimise-eta --help
+    - euphonic-show-sampling --help
+    - euphonic-intensity-map --help
+  requires:
+    - pip
+    - matplotlib >=2.0.0
```
Comment (xhochy) [inline recipes/euphonic/meta.yaml:56]: Why is this only required here? Also would `matplotlib-base` suffice (probably)?

File: recipes/euphonic/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+build:
+  number: 0
+  entry_points:
+    - euphonic-dispersion = euphonic.cli.dispersion:main
+    - euphonic-dos = euphonic.cli.dos:main
+    - euphonic-optimise-eta = euphonic.cli.optimise_eta:main
+    - euphonic-show-sampling = euphonic.cli.show_sampling:main
+    - euphonic-intensity-map = euphonic.cli.intensity_map:main
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: True  # [py<36]
```
Comment (xhochy) [inline recipes/euphonic/meta.yaml:22]: Not needed anymore, we only build for `>=36`
```suggestion
```

File: recipes/euphonic/meta.yaml (hunk 1 above)
Comment (ajjackson) [reply recipes/euphonic/meta.yaml:56]: matplotlib is an optional dependency for this package; if people only want to use it for fast interpolation of phonon data then it's a rather heavy dependency to drag in. It is however required for the command-line tools (which present plots), and therefore is needed for the tests of those tools.

Comment (ajjackson) [reply recipes/euphonic/meta.yaml:56]: Matplotlib-base should be fine, good shout.

**Pull request merged**